magprime/automated_emails.py

```diff
@@ -15,7 +15,7 @@ def __init__(self, event):
     SeasonSupporterEmail(_event)
 
 
-AutomatedEmail(Attendee, 'MAGFest schedule, maps, and other FAQs', 'precon_faqs.html', lambda a: days_before(7, c.EPOCH))
+AutomatedEmail(Attendee, 'MAGFest schedule, maps, and other FAQs', 'precon_faqs.html', lambda a: days_before(7, c.EPOCH) and (a.paid == c.HAS_PAID or a.paid == c.NEED_NOT_PAY or (a.group and a.group.amount_paid))
```

Comment:
Hmm, I didn't realize this until now, but I'm not entirely sure who should be included.

Like, what you have here omits people who have already been refunded, which happens to people like panelists.  It also omits people from need-not-pay groups.

So instead of trying to include everyone who should be included, let's just exclude the people who should definitely be excluded.  Which is probably just people who meet one or more of the following criteria:
- people whose ``badge_status`` is ``INVALID_STATUS`` or ``DEFERRED_STATUS``
- people whose ``paid`` is ``NOT_PAID``

In theory that could include people who aren't coming, e.g. people from unpaid groups.  In practice I think it's better to err on the side of including people who shouldn't obviously/definitely be excluded.

Caleb, can you update this to instead check the conditions I mention above?  (Thanks for jumping on these tickets so fast!)